botocore/endpoint.py

```diff
@@ -247,16 +248,17 @@ def create_endpoint(self, service_model, region_name=None, is_secure=True,
             final_endpoint_url = endpoint['uri']
         if not is_valid_endpoint_url(final_endpoint_url):
             raise ValueError("Invalid endpoint: %s" % final_endpoint_url)
-        return self._get_endpoint(
-            service_model, final_endpoint_url, verify, response_parser_factory)
 
-    def _get_endpoint(self, service_model, endpoint_url,
-                      verify, response_parser_factory):
-        endpoint_prefix = service_model.endpoint_prefix
-        event_emitter = self._event_emitter
-        return self._get_endpoint_complex(endpoint_prefix, endpoint_url,
-                                          verify, event_emitter,
-                                          response_parser_factory)
+        proxies = self._get_proxies(final_endpoint_url)
+        verify_value = self._get_verify_value(verify)
+        return Endpoint(
+            final_endpoint_url,
+            endpoint_prefix=service_model.endpoint_prefix,
+            event_emitter=self._event_emitter,
+            proxies=proxies,
+            verify=verify_value,
+            timeout=timeout,
```

Comment:
Probably not. They are now constructed in one place which is ClientCreator. The Endpoint, thanks to its underlying infrastructure "requests", supports timeout=None or timeout=integer or timeout=(connect_timeout, read_timeout) by nature.